be/src/olap/rowset_builder.cpp

```diff
@@ -362,7 +362,15 @@ void RowsetBuilder::_build_current_tablet_schema(int64_t index_id,
                                                     indexes[i], ori_tablet_schema);
     }
     if (_tablet_schema->schema_version() > ori_tablet_schema.schema_version()) {
-        _tablet->update_max_version_schema(_tablet_schema);
+        // After schema change should include extracted column
+        if (_tablet_schema->num_variant_columns() > 0) {
```

Comment:
Add detail case and comment why we need to add this logic

Reply:
done